### ISSUE TITLE
Complex python-monitor does not get custom config values

### ISSUE THREAD

**Issue by mattweber** [signalfx/signalfx-agent#1255]
Hello, I am writing a custom `python-monitor` and the `config` passed into the `configure` method does not have any of my additional configuration values I specified.

Agent Version:
```
agent-version: 5.0.1, collectd-version: 5.8.0-sfx0
```

Monitor Config:
```
- type: python-monitor
  discoveryRule: 'port == 9200 && host == "127.0.0.1"'
  scriptFilePath: /etc/signalfx/custom/es-extras.py
  intervalSeconds: 60
  username: admin
  password: admin
  useHTTPS: true
  skipVerify: true
```

The `config` object that I get passed in `configure` method:
```
{'MonitorID': '12', 'ValidateDiscoveryRule': False, 'configEndpointMappings': None, 'datapointsToExclude': [], 'dimensionTransformations': None, 'disableEndpointDimensions': False, 'disableHostDimensions': False, 'discoveryRule': 'port == 9200 && host == \"127.0.0.1\"', 'extraDimensions': None, 'extraDimensionsFromEndpoint': None, 'extraGroups': None, 'extraMetrics': None, 'host': '127.0.0.1', 'intervalSeconds': 60, 'name': 'java', 'port': 9200, 'pythonBinary': '/usr/lib/signalfx-agent/bin/python', 'pythonPath': None, 'scriptFilePath': '/etc/signalfx/custom/es-extras.py', 'solo': False, 'type': 'python-monitor'}
```

As you can see it is missing `username`, `password`, `useHTTPS`, and `skipVerify`.

**Comment by jchengsfx** [comment]
@mattweber To help isolate the issue, can you try to reconfigure the monitor without `discoveryRule` and use the `host` and `port` options instead?

**Comment by mattweber** [comment]
@jchengsfx thanks for the quick response, when I use `host` and `port` the config I get is correct.

**Comment by keitwb** [comment]
This should be fixed by #1306.  The way the discovered endpoint was interacting with the inline config in this monitor was causing the python-monitor config to be clobbered.

**Comment by keitwb** [comment]
Fixed by #1306 and released in 5.1.5.
